src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -457,6 +462,127 @@ public RealJenkinsRule prepareHomeLazily(boolean prepareHomeLazily) {
         return this;
     }
 
+    /**
+     * Use {@link #withFIPSEnabled(Path)} with default value of {@code target/dependency}
+     */
+    public RealJenkinsRule withFIPSEnabled() {
+        return withFIPSEnabled(Path.of("target", "dependency"));
+    }
+
+    /**
+     * Enable FIPS mode assuming all FIPS BouncyCastle jars (bc-fips.jar, bctls-fips.jar and bcpkix-fips.jar) are in the provided directory.
+     * This will set some System properties such :
+     * <ul>
+     *   <li>-Xbootclasspath with FIPS BouncyCastle jars</li>
+     *   <li>java.security.properties to a new temp file configuring the security provider to FIPS BouncyCastle</li>
+     *   <li>-Dsecurity.overridePropertiesFile=true</li>
+     *   <li>-Djavax.net.ssl.trustStoreType=PKCS12</li>
+     *   <li>-Djenkins.security.FIPS140.COMPLIANCE=true</li>
+     *   <li>-Dcom.redhat.fips=false</li>
+     * </ul>
+     * @param fipsLibrariesPath path to directory with FIPS BouncyCastle jars
+     */
+    public RealJenkinsRule withFIPSEnabled(@NonNull Path fipsLibrariesPath) {
+        this.fipsLibrariesPath = Objects.requireNonNull(fipsLibrariesPath);
+        // check expected BouncyCastle fips jars are here
```

Comment:
discussion started here https://github.com/jenkinsci/jenkins-test-harness/pull/825#discussion_r1739610865

Reply:
> [olamy](https://github.com/olamy) [5 days ago](https://github.com/jenkinsci/jenkins-test-harness/pull/825#discussion_r1739610865)
another option would have been to use maven-resolver API and download files for the user. But this would mean we force the jars version here.

> @[jtnord](https://github.com/jtnord) jtnord [yesterday](https://github.com/jenkinsci/jenkins-test-harness/pull/825#discussion_r1743379921)
the java security and version of the jars may well be intertwined, so I am not sure that forcing the versions here would be a bad thing.
Additionally with the current setup all users need to duplicate the BC FIPS libraries in their pom config?

> @[olamy](https://github.com/olamy) olamy [7 hours ago](https://github.com/jenkinsci/jenkins-test-harness/pull/825#discussion_r1744688493)
Yes but at least users have the freedom to eventually test some upgrades to new version (note it's in the plugin configuration so hopefully dependabot or other tools will not propose any upgrade)
The problem with using something to download dependencies (because dependencies may not be in the local repository) is that all the setup must be available here (such as the path to the used settings.xml, which can be different from ~/.m2/settings.xml), cli request properties, the content of .mvn/maven.config)
That makes something complicated to achieve and maintain (as Maven3 and Maven4 might have differences) for minimal gain.
Here, we provide a pom snippet to copy/paste. We can upgrade the pom snippet if we need to update the recommended versions.

Reply:
This would be a double edged sword.
On the one hande this may be nice for plugin authors, but on the other tools like the PCT would not be able to check for regressions in a sweeping manner by just bumping the test-harness.
Newer versions (e.g. the 2.x line) of BC-FIPS *may* (I have no idea) require different changes to the `java.security` file.

Reply:
> This would be a double edged sword. On the one hande this may be nice for plugin authors, but on the other tools like the PCT would not be able to check for regressions in a sweeping manner by just bumping the test-harness. Newer versions (e.g. the 2.x line) of BC-FIPS _may_ (I have no idea) require different changes to the `java.security` file.

at least it works for the current version we marked in core as named `FIPS140`
the release here are quick and cheap  why trying to solve we still have no ideas if it will be a problem or not.

Reply:
my concern is that pushing this down to plugins adds technical debt.  (e.g. rather than just being able to bump a jenkins-test-harness dep (possibly via a pom update) you end up needing to change the libraries and the test-harness in lock-step and once that is used in numerous plugins is more work that we would want it to be. (and changing how the test harness gets the libraries would similarly be more work that we would want it to be).

So it was a question of "are we definiately going to have this issue where a plugin can not just bump the bcfips libraries for testing because it also needs changes to the `java.security` file (in which case the way we have it today would be wrong and imply that is could indeed work.

I managed to track down the 2.0.0.0 [porting guide](https://downloads.bouncycastle.org/fips-java/docs/BC-FJA%202.0.0%20Porting%20Guide.pdf) and the [user guide](https://downloads.bouncycastle.org/fips-java/docs/BC-FJA-UserGuide-2.0.0.pdf) so I think this concern does not matter, witht he exception that for 2.0 a new library is needed in addition to the existing set.

Reply:
Perfectly understandable.
What about having another artifact bundling all together the fips bc libraries we know working together for a version x of the specifications? And so we can reuse that here eventually (unpacking them for users of the testing tool) having some constants/enum which can be used to generate the content of the security file?

Reply:
> so I think this concern does not matter, with the exception that for 2.0 a new library is needed in addition to the existing set.

which is actually a concern, so not sure why I wrote that!

> What about having another artifact bundling all together the fips bc libraries we know working together for a version x of the specifications?

I think there are a few things that we want to think about here.

1) how can we update the libraries to test a plugin automatically (e.g. during a PCT execution of a plugin (this could be bumping a J-T-H version which has the newer versions, but newer j-t-h's are not always backwards compatable).
2) how can we change what libraries are used (e.g. add the new library needed for BC-FIPS 2.0.0.0)
3) how can we make it easy for plugins to get the "right combination" (if plugins are to do this not the j-t-h)
4) how can we do this without introducing debt we need to maintain.

I think 2 can be solved by just looking for *all* libraries in a well known location during the RealJenkinsRule startup.
(this punts the need to provide this to the plugin so violates 3)

Reply:
Right another solution. https://github.com/olamy/fips-bundle-test.
The generated jars will contain known working/certified/validated/right/etc.... combinations of BC-FIPS jars and generated `security.properties` according to those jars.
As explained, this used version can be overridden by an environment variable or system property; the PCT hook can certainly do this.
This makes this very simple to use (no need to add some complex pom fragments) 
If we need to add BC-FIPS 2.0.0.0, we simply need to add this new version to the bundle jar.

Reply:
that looks like a good idea.
I am also wondering if we can use the same method for the acceptance-test-harness to configure a JVM in a FIPSlike way.

Reply:
sure but only for WinstoneController :) (we can forget JBossController and TomcatController)